Tests/UpstreamIntegrationTests/util/AnyCodableUtils.swift

```diff
@@ -0,0 +1,666 @@
+                
+                    lhs: \#(String(describing: lhs))
+                    
+                    rhs: \#(String(describing: rhs))
+                    
+                    key path: \#(keyPathAsString(keyPath: keyPath))
+                """#, file: file, line: line)
+            }
+            return false
+        }
+        if lhs.count != rhs.count {
+            if shouldAssert {
+                XCTFail(#"""
+                    lhs and rhs counts do not match.
+                    lhs count: \#(lhs.count)
+                    rhs count: \#(rhs.count)
+                    
+                    lhs: \#(lhs)
+                    
+                    rhs: \#(rhs)
+                    
+                    key path: \#(keyPathAsString(keyPath: keyPath))
+                """#, file: file, line: line)
+            }
+            return false
+        }
+        var finalResult = true
+        for (index, valueTuple) in zip(lhs, rhs).enumerated() {
+            var keyPath = keyPath
+            keyPath.append(index)
+            finalResult = finalResult && assertEqual(lhs: valueTuple.0, rhs: valueTuple.1, keyPath: keyPath, file: file, line: line, shouldAssert: shouldAssert)
+        }
+        return finalResult
+    }
+    
+    /// Performs testing assertions between two `[AnyCodable]` instances.
+    static func assertFlexibleEqual(validation: [AnyCodable]?, input: [AnyCodable]?, keyPath: [Any], exactMatchTree: [String: Any]?, file: StaticString = #file, line: UInt = #line, shouldAssert: Bool = true) -> Bool {
+        if validation == nil {
+            return true
+        }
+        guard let lhs = validation, let rhs = input else {
+            if shouldAssert {
+                XCTFail(#"""
+                    Validation JSON is non-nil but input JSON is nil.
+                
+                    validation: \#(String(describing: validation))
+                    
+                    input: \#(String(describing: input))
+                    
+                    key path: \#(keyPathAsString(keyPath: keyPath))
+                """#, file: file, line: line)
+            }
+            return false
+        }
+        if lhs.count > rhs.count {
+            if shouldAssert {
+                XCTFail(#"""
+                    Validation JSON has more elements than input.
+                    validation count: \#(lhs.count)
+                    input count: \#(rhs.count)
+                    
+                    validation: \#(lhs)
+                    
+                    input: \#(rhs)
+                    
+                    key path: \#(keyPathAsString(keyPath: keyPath))
+                """#, file: file, line: line)
+            }
+            return false
+        }
+        // this is where the craziness begins
+        // 1 check if exact match tree has [*] or [*0,1,2,3,4,...]
+            // [*] means only the first element from validation is taken into account, and strongly matched
```

Comment:
why does * mean first element? usually * means all. Why not give` 1, 2, or specific positions` for exact matching specific element?